satellite/gracefulexit/validation.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gracefulexit
+
+import (
+	"bytes"
+	"context"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/signing"
+)
+
+func verifyPieceTransferred(ctx context.Context, message *pb.StorageNodeMessage_Succeeded, transfer *pendingTransfer) error {
+	if transfer.satelliteMessage == nil {
+		return Error.New("Satellite message cannot be nil")
+	}
+	if transfer.satelliteMessage.GetTransferPiece() == nil {
+		return Error.New("Satellite message transfer piece cannot be nil")
+	}
+	if transfer.satelliteMessage.GetTransferPiece().GetAddressedOrderLimit() == nil {
+		return Error.New("Addressed order limit on transfer piece cannot be nil")
+	}
+	if transfer.satelliteMessage.GetTransferPiece().GetAddressedOrderLimit().GetLimit() == nil {
+		return Error.New("Addressed order limit on transfer piece cannot be nil")
+	}
+	if transfer.path == nil {
+		return Error.New("Transfer path cannot be nil")
+	}
+
+	originalOrderLimit := message.Succeeded.GetOriginalOrderLimit()
+	if originalOrderLimit == nil {
+		return ErrInvalidArgument.New("Original order limit cannot be nil")
+	}
+	originalPieceHash := message.Succeeded.GetOriginalPieceHash()
+	if originalPieceHash == nil {
+		return ErrInvalidArgument.New("Original piece hash cannot be nil")
+	}
+	replacementPieceHash := message.Succeeded.GetReplacementPieceHash()
+	if replacementPieceHash == nil {
+		return ErrInvalidArgument.New("Replacement piece hash cannot be nil")
+	}
+
+	// verify that the original piece hash and replacement piece hash match
+	if !bytes.Equal(originalPieceHash.Hash, replacementPieceHash.Hash) {
+		return ErrInvalidArgument.New("Piece hashes for transferred piece don't match")
+	}
+
+	// verify that the satellite signed the original order limit
+	err := endpoint.orders.VerifyOrderLimitSignature(ctx, originalOrderLimit)
```

Comment:
you probably see this in jenkins but can't access this endpoint from this function